src/Subtractor.js

```diff
@@ -33,45 +33,88 @@ class Subtractor extends Observable {
     this._polyphony = 1
     this._detune = 0
 
+    this._voices = 4
+    this._glide = 1
+
+    this._activeNotes = {};
+
     this.masterGain = this.context.createGain()
 
     // static connections
     this.filter2.filter.connect(this.masterGain)
     this.masterGain.connect(this.context.destination)
 
+    this.pipeline = this.pipeline.bind(this)
+
     // only perform certain tasks once the DOM is ready
     document.addEventListener('DOMContentLoaded', () => {
       this.startOscilloscope()
       this.loadPreset({})
+      this.osc1.notifyObservers()
+      this.osc2.notifyObservers()
     })
   }
+  
+  moveNote(n1, n2) {
+    const voices = this._activeNotes[n1]
+
+    Object.keys(voices)
+      .filter(i => voices[i])
+      .forEach((voice) => {
+        Object.keys(voice).forEach((v) => {
+          voices[voice[v]].move(
+            n2, 
+            this._polyphony, 
+            this._detune, 
+            this.context.currentTime + knobToSeconds(this._glide)
+          )
+        })
+      })
+      
+    renameObjectKey(this._activeNotes, n1, n2);
+  }
 
   noteOn(note) {
-    let returnOscs = [this.osc1, this.osc2].map((osc) => {
-      if (!osc.enabled) { 
-        return []
-      }
-      return osc.start(note, this._polyphony, this._detune)
-                .map(startedOsc => this.pipeline(startedOsc))
-    })
-
-    // osc.start returns an array of osc references, 
-    return [].concat(...returnOscs)
+    const activeNoteKeys = Object.keys(this._activeNotes)
+
+    if (activeNoteKeys.length >= this._voices) {
+      this.moveNote(activeNoteKeys[0], note);
```

Comment:
ahh, we can do some distance logic here to get more dynamic behavior if/when desired

Reply:
@stealthybox I agree. But there is a difficult problem ahead, consider this:

In an analog synth there is a maximum voice count, it is not possible to be hearing more oscillators available past max voices simultaneously.

This synth tries to model that analog functionality by limiting the amount of `Osc` instances playing at once. Currently this is done by tracking the number of currently playing notes as `Subtractor._activeNotes` in this PR. But a more advanced implementation would involve tracking the number of active oscillators, regardless of note on/off status - this is a larger problem to solve though.

Reply:
If we look at the voices as grouped together, perhaps we should slide all notes the same distance.
This could inspire some UI refactoring to reflect the state of the effective notes in addition to the keyboard.

NJ on this -- it's sooper cool 😎